game_patch/multi/kill.cpp

```diff
@@ -150,6 +152,11 @@ void on_player_kill(rf::Player* killed_player, rf::Player* killer_player)
         }
 
         multi_apply_kill_reward(killer_player);
+
+        // Spectate killer if follow killer is on and we weren't involved
+        if (g_spectate_mode_follow_killer && killed_player != rf::local_player && killer_player != rf::local_player) {
```

Comment:
Imo should be in `multi_spectate_on_player_death`.

Reply:
I originally had it there, I moved it specifically because I thought it would be best practice and more lightweight to do the checks here before falling the function.

Why do you think it should be in `multi_spectate_on_player_death`?

Reply:
It is poor structure. It lacks a clear separation and it leads to hard-to-maintain code.